src/alloc.c

```diff
@@ -617,6 +617,13 @@ vim_free(void *x)
     }
 }
 
+    void
+vim_free_string(string_T *s)
+{
+    VIM_CLEAR(s->string);
+    s->length = 0;
+}
```

Comment:
Maybe this can be changed to a macro instead?
```c
#define VIM_CLEAR_STRING(s) \
    do { \
	VIM_CLEAR(s.string); \
	s.length = 0; \
    } while (0)
```

Reply:
Ok.